utils/priority_queue/priority-queue.go

```diff
@@ -32,13 +33,13 @@ func (pq *wrapper[V, P]) Swap(i, j int) {
 	(*pq)[j].index = j
 }
 
-func (pq *wrapper[V, P]) Push(x *item[V, P]) {
-	item := x
+func (pq *wrapper[V, P]) Push(x any) {
+	item := x.(*Item[V, P])
 	item.index = len(*pq)
 	*pq = append(*pq, item)
 }
 
-func (pq *wrapper[V, P]) Pop() *item[V, P] {
+func (pq *wrapper[V, P]) Pop() any {
```

Comment:
Why did you change it to any?

Reply:
`container/heap` functions require this to implement this interface. It doesn't accept anything else :/

```go
type Interface interface {
	sort.Interface
	Push(x any) // add x as element Len()
	Pop() any   // remove and return element Len() - 1.
}
```